pkg/querier/queryrange/codec.go

```diff
@@ -173,38 +173,76 @@ func (r *LokiSeriesRequest) LogToSpan(sp opentracing.Span) {
 
 func (*LokiSeriesRequest) GetCachingOptions() (res queryrangebase.CachingOptions) { return }
 
-func (r *LokiLabelNamesRequest) GetEnd() int64 {
-	return r.EndTs.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+// In some other world LabelRequest could implement queryrangebase.Request.
+type LabelRequest struct {
+	path string
```

Comment:
The path needs to be save for backwards compatibility. This refactoring will help to extract the `logproto.LabelRequest`. See https://github.com/grafana/loki/pull/10688/files#diff-48228ecb9d835b591a8af90bff973f0acb4f1fdd984f5fa1e6926bb4b263981aR63-R75